spinta/manifests/mermaid/helpers.py

```diff
@@ -0,0 +1,290 @@
+        if self.is_enum:
+            class_text = f"{class_text}<<enumeration>>\n"
+        properties_text = "".join(f"{str(prop)}\n" for prop in self.properties)
+        class_text += properties_text
+        class_text += "}"
+        return class_text
+
+
+@dataclass
+class MermaidProperty:
+    name: str
+    access: Access | None = None
+    type: str | None = None
+    cardinality: bool | None = None
+    multiplicity: str | None = None
+
+    ACCESS_MAPPING = {
+        Access.open: '+',
+        Access.public: '#',
+        Access.protected: '~',
+        Access.private: '-'
+    }
+
+    def __str__(self):
+
+        property_string = self.name
+
+        if self.access is not None:
+            access = self.ACCESS_MAPPING[self.access]
+            property_string = f"{access} {property_string}"
+
+        if self.type is not None:
+            property_string = f"{property_string} : {self.type}"
+
+        if self.cardinality or self.multiplicity:
+            property_string = f"{property_string} [{int(self.cardinality) if self.cardinality is not None else ''}..{self.multiplicity if self.multiplicity is not None else ''}]"
+
+        return property_string
+
+
+class RelationshipDirection(enum.Enum):
+    FORWARD = 'forward'
+    BACKWARD = 'backward'
+
+
+class RelationshipType(enum.Enum):
+    ASSOCIATION = 'association'
+    DEPENDENCY = 'dependency'
+    INHERITANCE = 'inheritance'
+
+
+@dataclass
+class MermaidRelationship:
+    node1: str
+    node2: str
+    type: RelationshipType
+    direction: RelationshipDirection = RelationshipDirection.FORWARD
+    cardinality: bool = False
+    multiplicity: str = ""
+    label: str = ""
+
+    def __str__(self):
+
+        if self.direction == RelationshipDirection.FORWARD:
+            if self.type == RelationshipType.ASSOCIATION:
+                arrow = '-->'
+            if self.type == RelationshipType.DEPENDENCY:
+                arrow = '..>'
+            if self.type == RelationshipType.INHERITANCE:
+                arrow = '--|>'
+        else:
+            if self.type == RelationshipType.ASSOCIATION:
+                arrow = '<--'
+            if self.type == RelationshipType.DEPENDENCY:
+                arrow = '<..'
+            if self.type == RelationshipType.INHERITANCE:
+                arrow = '<|--'
+
+        if self.cardinality or self.multiplicity:
+            cardinality_multiplicity = f' "[{int(self.cardinality)}..{self.multiplicity}]"'
+        else:
+            cardinality_multiplicity = ""
+
+        relationship_text = f"{self.node1} {arrow}{cardinality_multiplicity} {self.node2}"
+        if self.label:
+            relationship_text += f" : {self.label}"
+
+        return relationship_text
+
+
+def find_backref_property(models, model_name):
+    for name, model in models.items():
+        for prop in model.get_given_properties().values():
+            if prop.type == 'backref' and prop.model.name == model_name:
```

Comment:
Try to avoid dynamic code dependencies, because they are difficult to manage. Instead use static dependencies, in this case `isinstance(prop.dtype, BackRef)`.

Reply:
pakeičiau visur